contributors/devel/development.md

```diff
@@ -279,20 +279,42 @@ To build binaries for all platforms:
 make cross
 ```
 
+#### Install etcd
+```
+hack/install-etcd.sh
+```
+
+#### Verification
+```
+# Run all hack/verify-*.sh scripts
+make verify
+```
+
 #### Test
 
 ```sh
 cd $working_dir/kubernetes
 
+# Run all hack/update-*.sh scripts
+# For small change, run "make verify" and then relevant scripts, for example:
+# hack/update-gofmt.sh (when adding new files)
+# hack/update-bazel.sh (when updating dependencies / imports)
+make update
+
 # Run every unit test
 make test
 
 # Run package tests verbosely
 make test WHAT=pkg/util/cache GOFLAGS=-v
+# Or use bazel (quote "..." when using zsh)
+bazel test //pkg/util/cache/...
 
 # Run integration tests, requires etcd
 make test-integration
 
+# (Optional) Build e2e test binaries
+make all WHAT=test/e2e/e2e.test
+
 # Run e2e tests
 make test-e2e
```

Comment:
`make test-e2e` = `go run hack/e2e.go -- -v --build --up --test --down` = build test binaries, turn up a test cluster, run all the tests, and tear the cluster down. 
That takes a LONG time. Contributors should only run the tests related to the code they touched to save themselves time (so is the section about unit test and integration test). 

It'd be good to explain this and link to https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md#building-kubernetes-and-running-the-tests